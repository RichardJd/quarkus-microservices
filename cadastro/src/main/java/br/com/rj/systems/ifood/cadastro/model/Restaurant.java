package br.com.rj.systems.ifood.cadastro.model;

import io.quarkus.hibernate.orm.panache.PanacheEntityBase;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToOne;
import javax.persistence.Table;
import java.time.OffsetDateTime;

@Entity
@Table(name = "restaurant")
public class Restaurant extends PanacheEntityBase {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long id;

    public String owner;

    public String cnpj;

    public String name;

    @OneToOne(cascade = CascadeType.ALL)
    public Localization localization;

    @CreationTimestamp
    public OffsetDateTime createdDate;

    @UpdateTimestamp
    public OffsetDateTime updatedDate;
}
