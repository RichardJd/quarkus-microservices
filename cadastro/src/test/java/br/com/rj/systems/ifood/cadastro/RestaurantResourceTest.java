package br.com.rj.systems.ifood.cadastro;

import br.com.rj.systems.ifood.cadastro.model.Restaurant;
import com.github.database.rider.cdi.api.DBRider;
import com.github.database.rider.core.api.configuration.DBUnit;
import com.github.database.rider.core.api.configuration.Orthography;
import com.github.database.rider.core.api.dataset.DataSet;
import io.quarkus.test.common.QuarkusTestResource;
import io.quarkus.test.junit.QuarkusTest;
import io.restassured.RestAssured;
import io.restassured.http.ContentType;
import org.approvaltests.JsonApprovals;
import org.junit.Assert;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import javax.ws.rs.core.Response;

@DBRider
@DBUnit(caseInsensitiveStrategy = Orthography.LOWERCASE)
@QuarkusTest
@QuarkusTestResource(RegisterTestLifecycleManager.class)
class RestaurantResourceTest {

    @Test
    @DataSet("restaurants-scenario-1.yml")
    void testFindRestaurants() {
        var result = RestAssured.given()
                .when().get("/restaurants")
                .then()
                .statusCode(200)
                .extract().asString();

        JsonApprovals.verifyJson(result);
    }

    @Test
    @DataSet("restaurants-scenario-1.yml")
    void testUpdateRestaurant() {
        var dto = new Restaurant();
        dto.name = "newName";
        var parameterValue = 123L;

        RestAssured.given()
                .contentType(ContentType.JSON)
                .with().pathParam("id", parameterValue)
                .body(dto)
                .when().put("/restaurants/{id}")
                .then()
                .statusCode(Response.Status.NO_CONTENT.getStatusCode())
                .extract().asString();

        Restaurant findById = Restaurant.findById(parameterValue);

        Assertions.assertEquals(dto.name, findById.name);
    }
}