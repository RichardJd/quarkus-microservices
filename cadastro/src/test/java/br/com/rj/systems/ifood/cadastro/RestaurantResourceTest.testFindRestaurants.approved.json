[
  {
    "id": 123,
    "name": "New Restaurant",
    "owner": "ID do KeyClock"
  },
  {
    "id": 234,
    "name": "New Restaurant 2",
    "owner": "ID do KeyClock 2"
  }
]